src/connection.rs

```diff
@@ -180,7 +180,6 @@ pub enum ConnectionState {
 impl From<i64> for ConnectionState {
     fn from(state: i64) -> Self {
         match state {
-            0 => ConnectionState::Unknown,
```

Comment:
Maybe just a `warn!(...)` log entry? Otherwise it will complicate the code too much and I doubt we will even encounter this scenario - unless NM API adds new states in the future, which we can treat as unknown until we add them.